msgpack/src/main/scala/fs2/data/msgpack/low/internal/ItemSerializer.scala

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2024 fs2-data Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2
+package data
+package msgpack
+package low
+package internal
+
+import scodec.bits._
+
+private[low] object ItemSerializer {
+  class MalformedItemError extends Error("item exceeds the maximum size of it's format")
+  class MalformedStringError extends MalformedItemError
+  class MalformedBinError extends MalformedItemError
+  class MalformedIntError extends MalformedItemError
+  class MalformedUintError extends MalformedItemError
+
+  /** Checks whether integer `x` fits in `n` bytes. */
+  @inline
+  private def fitsIn(x: Int, n: Long): Boolean =
+    java.lang.Integer.compareUnsigned(x, (Math.pow(2, n.toDouble).toLong - 1).toInt) <= 0
```

Comment:
If this is really necessary, see comment below, then I believe we can improve its performance by using a bit shift. Taking both arguments as `Long`, that'd be 
```scala
java.lang.Long.compareUnsigned(x, (1L << n) - 1).toInt) <= 0
```

(needs some testing though, I'm no bit shift pro and probably got an off-by-one somewhere)

Reply:
I completely got rid of the `fitsIn` function as we don't use `Int`s in this context anymore.